engines/registry/lib/registry/engine.rb

```diff
@@ -2,5 +2,17 @@ module Registry
   class Engine < ::Rails::Engine
     isolate_namespace Registry
     config.generators.api_only = true
+
+    config.after_initialize do
+      Api::Connect::V3::Systems::ActivationsController.class_eval do
+        before_action :handle_cache, only: %w[index]
+
+        def handle_cache
+          unless request.headers['X-Instance-Data'] && ZypperAuth.verify_instance(request, logger, @system, registry: true)
```

Comment:
I think it makes sense to push the check for the existence of the header `X-Instance-Data` to the `verify_instance` code. The `verify_instance` code assumes the header is there but should have a shortcut `false` exit condition if it doesn't exist, IMHO.

Reply:
I like the change you suggest, that implies...that _all_ the calls/requests that pass through `ZypperAuth.verify_instance` *must* have the header set, which is fine by me but implies 
a few changes more than updating that line on registry engine

Reply:
Updated